Recruit-Training/Embedded-Recruit-Training/Core/Src/GreenLedToggleTask.c

```diff
@@ -7,11 +7,39 @@ void greenLedToggleTask(void const* arg)
 
     osMutexId_t* canMutex = (osMutexId_t*)arg;
 
+    static const uint32_t GREEN_LED_STATUS_STDID = 0xDDD;
+    static const uint32_t GREEN_LED_TOGGLE_FREQ = 100;
+
     for (;;)
     {
         //TODO: Add GREEN_LED_TOGGLE_FREQ to prevWakeTime
+        prevWakeTime += GREEN_LED_TOGGLE_FREQ;
         osDelayUntil(prevWakeTime);
-        //TODO: Check blue toggel flag and toggle blue LED
+
+        //TODO: Check green toggle flag and toggle green LED
+        if(greenLedToggleFlag) 
+        {
+            HAL_GPIO_TogglePin(GPIOA, LED_GREEN_Pin);
+        } 
+
         //TODO: Send CAN message indicating current state of LED
+        uint8_t GREEN_LED_STATUS_ID = HAL_GPIO_ReadPin(GPIOA, LED_GREEN_Pin);
+
+        if(osMutexAcquire(canMutex, 0) == osOK)
+        {
+            uint32_t mailboxVariable = HAL_CAN_GetTxMailboxesFreeLevel(&hcan2);
+            if(mailboxVariable != 0U)
+            {
+                canHeader.StdId = GREEN_LED_STATUS_STDID;
+                canHeader.DLC = 1;
+                uint8_t GREEN_LED_MESSAGE[1] = {(~GREEN_LED_STATUS_ID) & 0b0000000};
```

Comment:
Just negating the LED status will get you the same thing. Since when you bitwise and with 0 you just get whatever is the original value.